TypeCobol/Compiler/SqlScanner/SqlScanner.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.Scanner;
+
+namespace TypeCobol.Compiler.SqlScanner
+{
+    /// <summary>
+    /// A Sql Scanner related to a Scanner 
+    /// </summary>
+    public static class SqlScanner
+    {
+        /// <summary>
+        /// True if the given char can bepart of an SQL keyword.
+        /// </summary>
+        /// <param name="c">The character to test</param>
+        /// <returns>True if yes, no otherwise</returns>
+        public static bool IsSqlKeywordPart(char c)
+        {
+            return c == '-' || Char.IsLetter(c);
+        }
+
+        public static Token ScanSqlKeywordOrExecStatementText(ref int currentIndex, int startIndex, int endIndex, int lastIndex, string line, TokensLine tokensLine)
+        {
+            int curIndex = currentIndex;
+            if (IsSqlKeywordPart(line[startIndex]))
+            {
+                for (; curIndex <= lastIndex && IsSqlKeywordPart(line[curIndex]); curIndex++) { }
+                int curEndIndex = curIndex - 1;
+                string text = line.Substring(startIndex, curEndIndex - startIndex + 1);
+                if (text.Equals("COMMIT", StringComparison.OrdinalIgnoreCase)) {
+                    currentIndex = curIndex;
+                    return new Token(TokenType.COMMIT, startIndex, curEndIndex, tokensLine);
+                }                    
+            }
+            // Consume all chars
+            currentIndex = endIndex + 1;
+            return new Token(TokenType.ExecStatementText, startIndex, endIndex, tokensLine);
+        }
```

Comment:
- the `currentIndex` param is an `out` param in fact
- no need for both `lastIndex` and `endIndex`, `endIndex` from the caller is the interesting value

```suggestion
public static Token ScanSqlKeywordOrExecStatementText(int startIndex, int endIndex, string line, TokensLine tokensLine, out int currentIndex)
        {
            int curIndex = startIndex;
            if (IsSqlKeywordPart(line[curIndex]))
            {
                for (; curIndex <= endIndex && IsSqlKeywordPart(line[curIndex]); curIndex++) { }
                int curEndIndex = curIndex - 1;
                string text = line.Substring(startIndex, curEndIndex - startIndex + 1);
                if (text.Equals("COMMIT", StringComparison.OrdinalIgnoreCase)) {
                    currentIndex = curIndex;
                    return new Token(TokenType.COMMIT, startIndex, curEndIndex, tokensLine);
                }                    
            }

            // Consume all chars as ExecStatementText
            currentIndex = endIndex + 1;
            return new Token(TokenType.ExecStatementText, startIndex, endIndex, tokensLine);
        }
```

Reply:
Integrated in [7f2afb7](https://github.com/TypeCobolTeam/TypeCobol/pull/1928/commits/7f2afb778747d8f26ff0ce751781c2ab8845aa78)